src/Components/Search/__tests__/SearchBar.test.tsx

```diff
@@ -0,0 +1,44 @@
+import Input from "Components/Input"
+import { SearchBar } from "Components/Search/SearchBar"
+import { MockBoot } from "DevTools/MockBoot"
+import { mount } from "enzyme"
+import React from "react"
+
+const searchResults = {
+  search: {
+    edges: [
+      {
+        node: {
+          displayLabel: "Percy Z",
+          href: "/cat/percy-z",
+          searchableType: "Cat",
+          id: "percy-z",
+        },
+      },
+    ],
+  },
+}
+
+describe("SearchBar", () => {
+  const getWrapper = viewer => {
+    return mount(
+      <MockBoot breakpoint="xl">
+        <SearchBar
+          viewer={viewer as any}
+          relay={{ environment: {}, refetch: () => null } as any}
+        />
+      </MockBoot>
+    )
+  }
+
+  it("displays search results", async () => {
+    const component = getWrapper(searchResults)
+    component
+      .find(Input)
+      .props()
+      .onChange({ target: { value: "percy" } } as any)
```

Comment:
No matter how hard i tried to type this properly `HTMLInputElement` I couldn't get it to not complain about all the _other_ missing attributes for that type. I just wanted the value anyway.

Reply:
This is also where doing `simulate("focus")` and `simulate("change", { target: { value: ...)` and various combinations didn't get the suggestions to render, leading to that hack above.

Reply:
Might be worth digging into recent Reaction PR's to look up @ds300's `simulateTyping` helper.

Reply:
Yea, I saw that. Any combination of those methods (click/focus, setting value, etc.) doesn't cause the suggestion lib to actually render the suggestions. It's possibly something in the way that lib works? That's definitely the right way to simulate the methods. That's why I had to go with the `alwaysRenderSuggestions={typeof jest === "undefined" ? false : true}` hack. This lib is really frustrating me to test!

Reply:
If there are async things happening in the library, we might need to "wait" for them before things get rendered.

[Here are some tests](https://github.com/artsy/reaction/blob/2a64f4028994c40e2595ada34b78e1ea9bc76d28/src/Apps/Order/Routes/__tests__/Payment.test.tsx#L105) that wait for async things to happen. 

Following the trail of that `update` function to [here](https://github.com/pepopowitz/reaction/blob/2a64f4028994c40e2595ada34b78e1ea9bc76d28/src/DevTools/RootTestPage.tsx#L17), here's what's happening: 

1. Wait for a 0-length promise to resolve, then
2. Call `update()` on the `root`, which is really just the top-level wrapper returned by `enzyme#mount`, then
3. Call `update()` on a wrapper that ...I'm not sure on this part, exactly. It might be an element that got selected during the test. 

Regardless of #3, I think it's worth trying 1&2 to see if asynchrony is the problem.

Reply:
In general, I feel like using `.simulate('change')` is the right move. Using `.props().onChange` seems like a workaround for enzyme problems, and that strikes me as something that will lead to future maintenance headaches.

Reply:
Int terms of waiting, there is also the [waitFor](https://github.com/artsy/reaction/blob/master/src/DevTools/enzyme%2BrenderUntil.d.ts) util that you can tap into.

Reply:
No dice on the `update()` . I did switch back to `simulate` which works w/ the `alwaysRenderSuggestions` hack, so that's no issue.

Reply:
Alright we figured it out (sorta...). Needed a `setImmediate(` on the assertions in the test.